assets/javascripts/admintable.js

```diff
@@ -3,20 +3,34 @@ function updateTextArea(textArea) {
     textArea.style.height = Math.min(textArea.scrollHeight + 5, 300) + 'px';
 }
 
-function addAdminTableRow() {
-    // add new row
+function showAdminTableRow(row) {
     var adminTable = window.adminTable;
-    var newRow = adminTable.row.add(adminTable.emptyRow);
-    var newRowIndex = newRow.index();
-    adminTable.rowData[newRowIndex] = jQuery.extend({isEditing: true}, adminTable.emptyRow);
-    newRow.invalidate().draw();
 
     // set pagination to the page containing the new row
     var pageInfo = adminTable.page.info();
-    var rowPosition = adminTable.rows()[0].indexOf(newRowIndex);
+    var rowPosition = adminTable.rows()[0].indexOf(row.index());
     if (rowPosition < pageInfo.start || rowPosition >= pageInfo.end) {
         adminTable.page(Math.floor(rowPosition / adminTable.page.len())).draw(false);
     }
+
+    // scroll to the new row
+    $('html').animate({scrollTop: $(row.node()).offset().top}, 250);
+}
+
+function addAdminTableRow() {
+    var adminTable = window.adminTable;
+
+    // clear search
+    $('#test-suites_filter input').val('');
```

Comment:
The id of filter div in machine table is "machines_filter",  Does it need  the same treatment?

Reply:
You're right - I will fix it.